packages/api/src/controllers/experiment/attestation.ts

```diff
@@ -50,15 +58,64 @@ app.post("/", validatePost("attestation"), async (req, res) => {
   return res.status(201).json(attestationMetadata);
 });
 
-function verifySigner(message, signature): boolean {
-  const verifiedSigner = ethers.verifyTypedData(
-    DOMAIN,
-    TYPES,
-    message,
-    signature
-  );
+async function verifySignature(
+  message: any,
+  signature: string,
+  signatureType?: string
+): Promise<string> {
+  const verifiedEIP712 =
+    (signatureType === "eip712" || !signatureType) &&
+    verifyEIP712Signature(message, signature);
+  const verifiedFlow =
+    (signatureType === "flow" || !signatureType) &&
+    (await verifyFlowSignature(message, signature));
+
+  return verifiedEIP712 ? "eip712" : verifiedFlow ? "flow" : null;
+}
+
+function verifyEIP712Signature(message: any, signature: string): boolean {
+  try {
+    const verifiedSigner = ethers.verifyTypedData(
+      DOMAIN,
+      TYPES,
+      message,
+      signature
+    );
+
+    return verifiedSigner === message.signer;
+  } catch (e) {
```

Comment:
I wonder if these error messages won't be useful to the user at some point

Reply:
I think that no, the user will just get "invalid signature". Here you can't read much more data from this error message.